backend/src/useCases/ownership.ts

```diff
@@ -1,20 +1,63 @@
-import { User } from "../models/index.js";
-import { ownershipRepository } from "../repositories/index.js";
+import { Owner, User } from "../models/index.js";
+import { ownershipRepository, usersRepository } from "../repositories/index.js";
 
 const setUserAsOwner = async (user: User, cid: string) => {
-  await ownershipRepository.setUserAsOwner(cid, user.provider, user.id);
+  await ownershipRepository.setUserAsOwner(
+    cid,
+    user.oauthProvider,
+    user.oauthUserId
+  );
 };
 
 const setUserAsAdmin = async (user: User, cid: string) => {
-  await ownershipRepository.setUserAsAdmin(cid, user.provider, user.id);
+  await ownershipRepository.setUserAsAdmin(
+    cid,
+    user.oauthProvider,
+    user.oauthUserId
+  );
 };
 
 const setObjectAsDeleted = async (user: User, cid: string) => {
-  await ownershipRepository.setObjectAsDeleted(user.provider, user.id, cid);
+  await ownershipRepository.setObjectAsDeleted(
+    user.oauthProvider,
+    user.oauthUserId,
+    cid
+  );
+};
+
+const getOwners = async (cid: string): Promise<Owner[]> => {
+  const ownerships = await ownershipRepository.getOwnerships(cid);
+  const users = await Promise.all(
+    ownerships.map((e) =>
+      usersRepository.getUserByOAuthInformation(
+        e.oauth_provider,
+        e.oauth_user_id
+      )
+    )
+  );
+
+  if (users.some((user) => !user)) {
+    console.log("users", users);
+
+    throw new Error("Inconsistent database state");
+  }
+
+  const safeUsers = users.map((user) => user!);
+
+  return safeUsers.map((user, index) => ({
+    handle: user.handle,
+    role: ownerships[index].is_admin ? "admin" : "viewer",
```

Comment:
Could we move the roles into an enum somewhere that everything can access them, instead to have them hardcoded everywhere?
(and if we base our poc on google drive, then maybe we will want to add a commenter role in the future?)

Reply:
You're right

Reply:
For the moment, I'll use only "admin" and "viewer" for simplifying a bit the first version